### DISCUSSION TITLE
Change marker text color

### DISCUSSION THREAD

**Discussion by lcrodrigues** [patrykandpatrick/vico#680] [Questions]
Is there a way to change the marker text color? I'd like to change that `0.13` text color but I can't find a way to do it.
On the `Marker.kt` file, line 58, we can change the color for the label on `rememberTextComponent`, but the change doesn't seems to affect the marker text color.

![Screenshot_1714421429](https://github.com/patrykandpatrick/vico/assets/16669026/887a8c8c-870d-49fc-947e-7c335834898a)

**Comment by patrickmichalik** [comment]
Hello! `DefaultCartesianMarkerValueFormatter` color-codes _y_ values by default. `TextComponent#color` is then used for sums, brackets, and the like, as can be seen in the eighth sample chart. You can turn off color coding via `DefaultCartesianMarkerValueFormatter#colorCode`. For `Marker.kt`, these are the changes:

```diff
@@ -34,6 +34,7 @@
 import com.patrykandpatrick.vico.core.cartesian.Insets
 import com.patrykandpatrick.vico.core.cartesian.marker.CartesianMarker
 import com.patrykandpatrick.vico.core.cartesian.marker.DefaultCartesianMarker
+import com.patrykandpatrick.vico.core.cartesian.marker.DefaultCartesianMarkerValueFormatter
 import com.patrykandpatrick.vico.core.common.Dimensions
 import com.patrykandpatrick.vico.core.common.component.TextComponent
 import com.patrykandpatrick.vico.core.common.copyColor
@@ -62,6 +63,7 @@
             textAlignment = Layout.Alignment.ALIGN_CENTER,
             minWidth = TextComponent.MinWidth.fixed(40.dp),
         )
+    val valueFormatter = remember { DefaultCartesianMarkerValueFormatter(colorCode = false) }
     val indicatorFrontComponent = rememberShapeComponent(Shape.Pill, MaterialTheme.colorScheme.surface)
     val indicatorCenterComponent = rememberShapeComponent(Shape.Pill)
     val indicatorRearComponent = rememberShapeComponent(Shape.Pill)
@@ -80,6 +82,7 @@
     return remember(label, labelPosition, indicator, showIndicator, guideline) {
         object : DefaultCartesianMarker(
             label = label,
+            valueFormatter = valueFormatter,
             labelPosition = labelPosition,
             indicator = if (showIndicator) indicator else null,
             indicatorSizeDp = 36f,
```

**Reply by lcrodrigues** [reply]
@patrickmichalik It worked good, thanks!
Is it possible to do the same for 1.14.0 version?

**Reply by patrickmichalik** [reply]
You’re welcome, @lcrodrigues! It certainly is. The equivalent of `DefaultCartesianMarker` is `MarkerComponent`. Instead of a `valueFormatter` constructor parameter, `MarkerComponent` has a `labelFormatter` property. The equivalent of `DefaultCartesianMarkerValueFormatter` is `DefaultMarkerLabelFormatter`, which also has a `colorCode` constructor parameter.
